WordPress/gutenberg
# [RNMobile] Global styles provider

## Description
This is a very first thing from https://github.com/wordpress-mobile/gutenberg-mobile/issues/2147

This PR introduces a global style - block-level support mechanism to the mobile.

### Terms:
- `wrapperStyles` - styles that are applied on the block level
```

<p class="has-text-color" style="color:#4e4d0e">Color set in paragraph</p>

```
- `globalStyles` - styles that are passed down through editor and are overridden by block-level ones

Recently a lot of changes have been made in terms of `custom colors` and style in general. 
The color and the background color is added to the wrapper on the web. (other styles are added or will be added soon):
- https://github.com/WordPress/gutenberg/pull/21021/files#diff-2fc437498340ef224897c8cf939aa171R205
- https://github.com/WordPress/gutenberg/pull/21021/files#diff-2fc437498340ef224897c8cf939aa171R216

In this PR I use `getEditWrapperProps` to get wrapper styles and I pass them to the block component. Thanks to that we can get `background` or `color`(and other props like line-height) of the block (if set) from `wrapperProps` like here - https://github.com/WordPress/gutenberg/pull/21637/files#diff-b27dffd07183b62521c86ec1d84cad1dR226

In addition, I added `GlobalStylesContext` to pass styles down to the nested blocks and override needed values on the block level. So, we can get correct styles wherever we use the `GlobalStylesContext.Consumer`. 
We should pass flatten Theme (global) > User (edited global styles) > Specific styles per post as the initial value and then the style will be updated on the block level by concatenating global styles and wrapper style. On the web, these values are set as inline styles and are inherited by blocks out of the box so it's enough to set them there.

I'm not sure where it should be done on mobile. In the current implementation, I do that in `block.native.js` https://github.com/WordPress/gutenberg/pull/21637/files#diff-42b230fc8f23c8ce7322e2d6ff6d5df6R63-R66
That means the styles are updated on the block level for all blocks with no exceptions and in the edit file of the block we have correct values. 

The second place where it could be done is an edit file of each block that has inner blocks. Thanks to that we could decide if that block should override global styles for its inner block. It is a different behavior than the web so I decided to go with the first option. This approach would also require external developers of 3rd party block to think about that if they want to override styles or not.

In both approaches, we have to keep in mind that sometimes we shouldn't apply the global styles as a fallback.
Example:
Let's say we have a group block where we set the background color to red. The red background color shouldn't be applied in a button block that is inside that group. So, in that case, we should set the background only when is set in `wrapperProps` of that button block.

## How has this been tested?
- Create a nested structure with groups, paragraphs, media-text in the web editor (local)
- Set custom text/background colors
- Copy HTML and use it on mobile  or use this one below:
```

<div class="wp-block-group has-text-color" style="color:#a54b01"><div class="wp-block-group__inner-container">
<p>Hello :)</p>



<div class="wp-block-group has-text-color" style="color:#a9d994"><div class="wp-block-group__inner-container">
<p>From colorful paragraphs :)</p>



<div class="wp-block-group has-text-color" style="color:#140c52"><div class="wp-block-group__inner-container">
<p>Colors set in groups</p>



<div class="wp-block-group"><div class="wp-block-group__inner-container">
<p class="has-text-color" style="color:#4e4d0e">Color set in paragraph</p>
</div></div>
</div></div>
</div></div>



<div class="wp-block-media-text alignwide is-stacked-on-mobile has-text-color has-background" style="background-color:#4b9cbf;color:#e4dc7c"><figure class="wp-block-media-text__media"></figure><div class="wp-block-media-text__content">
<p class="has-large-font-size">media-text works as well</p>
</div></div>
</div></div>

```
- Colors should be applied to the paragraphs/media-text (the group doesn't support colors yet)

## Screenshots 
### Web
<img width="697" alt="Screenshot 2020-04-20 at 17 49 50" src="https://user-images.githubusercontent.com/16336501/79771738-7baf2980-832f-11ea-8c67-0d5b434c0900.png">

### Mobile
<img width="350" src="https://user-images.githubusercontent.com/16336501/79766037-bdd46d00-8327-11ea-935f-8b6419b060ce.png">

## Types of changes
Introduction of the mechanism that allows passing styles to the inner blocks and use `getEditWrapperProps` to get block-level styles. Global > Block > Block > ... (Global > Group > Group > Paragraph)

## Checklist:
